src/server/index.js

```diff
@@ -55,46 +61,30 @@ app.prepare()
             })
         );
 
-        logger.info("configuring passport");
-        server.use(passport.initialize());
-        server.use(passport.session());
+        logger.info("configuring keycloak");
+        server.use(keycloakClient.middleware());
 
         logger.info("adding the /login handler");
-        server.get(
-            "/login",
-            passport.authenticate("oauth2", {
-                session: true,
-                successReturnToOrRedirect: "/",
-            })
-        );
+        server.get("/login", keycloakClient.protect());
 
         logger.info("adding the /login/* handler");
-        server.get("/login/*", ensureLoggedIn("/login"), (req, res) => {
+        server.get("/login/*", keycloakClient.protect(), (req, res) => {
             res.redirect(req.url.replace(/^\/login/, ""));
         });
 
-        logger.info("adding the /logout handler");
-        server.get("/logout", (req, res) => {
```

Comment:
So keycloak doesn't support logging out?

Reply:
It does; the logout endpoint is built into the middleware, so we don't have to define it explicitly anymore.